src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -133,16 +133,40 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
                     collection: out collection);
             }
 
+            // TODO: in both nongeneric and generic forks below, we should unify both "custom tasklike" and "Task/Task<T>"
+            // down the same custom-tasklike path. (Should mscorlib ever device to put [AsyncBuilder] on Task then that
+            // will happen anyway).
+
             if (method.IsTaskReturningAsync(F.Compilation))
             {
                 var returnType = (NamedTypeSymbol)method.ReturnType;
                 NamedTypeSymbol builderType;
-                MethodSymbol createBuilderMethod;
-                PropertySymbol taskProperty;
-                bool customBuilder = returnType.IsCustomTaskType(out builderType, out createBuilderMethod);
+                MethodSymbol createBuilderMethod = null;
+                PropertySymbol taskProperty = null;
+
+                object builderArgument;
+                bool customBuilder = returnType.IsCustomTaskType(out builderArgument);
                 if (customBuilder)
                 {
-                    taskProperty = GetCustomTaskProperty(F, builderType);
+                    builderType = builderArgument as NamedTypeSymbol;
+                    if (builderType?.IsErrorType() == true)
+                    {
+                    }
```

Comment:
Definitely agree, the empty `if` block here seems off.  It's definitely something I'd poke at if I came across this code in the future.  

---
In reply to: [76644154](https://github.com/dotnet/roslyn/pull/13405#discussion_r76644154) [](ancestors = 76644154)